src/components/LanguageProvider.tsx

```diff
@@ -0,0 +1,40 @@
+import React, { createContext, useContext } from "react";
+import sv from "../language/sv.json";
+
+import languages from '../language/';
+
+import Storage from '../Storage'
+
+const LanguageContext = createContext(sv);
+
+/**
+ * Provides the current language dictionary to its children.
+ */
+export const LanguageProvider: React.FC = ({children}) => {
+  const [settings, modifySettings] = Storage.useSettings();
+  const dict = languages[settings.language];
+
+  return (
+    <LanguageContext.Provider value={dict}>
+      {children}
+    </LanguageContext.Provider>
+  );
+};
+
+/**
+ * Function returning the text from the correct language dictionary.
+ * @param entry keyvalue for the entry in the current language json file
+ * @returns translation/value for the current language
+ * 
+ * @example
+ * Get the welcome message for the current language
+ * getTranslation("valuesHeaderEvaluation");
+ * returns "Värdering" if swedish is the current language
+ * ```
+ * keyvalues can be found in the respective json, and new translations should
+ * be added in every file (src/language).
+ * 
+ * TODO: Fix types and replace any
+ * ```
+ */
+export const getTranslation = (entry: string) => useContext<any>(LanguageContext)[entry];
```

Comment:
I like the idea of this function, but there are some things that will have to be straightened out before merging.

* This function is a Hook (transitive property), so it should be named useTranslation.
* I think it is better if it just returns the whole dictionary since using several values in a single component would require using the hook more than once (which you generally want to avoid for various reasons.)